lib/vagrant-cloudstack/action/run_instance.rb

```diff
@@ -34,7 +35,7 @@ def call(env)
           @network          = CloudstackResource.new(domain_config.network_id, domain_config.network_name, 'network')
           @service_offering = CloudstackResource.new(domain_config.service_offering_id, domain_config.service_offering_name, 'service_offering')
           @disk_offering    = CloudstackResource.new(domain_config.disk_offering_id, domain_config.disk_offering_name, 'disk_offering')
-          @template         = CloudstackResource.new(domain_config.template_id, domain_config.template_name, 'template')
+          @template         = CloudstackResource.new(domain_config.template_id, domain_config.template_name || env[:machine].config.vm.box, 'template')
```

Comment:
I guess we need to go deep into vagrant with this, and make it somehow disregard the `config.vm.box` attribute, or somehow (not sure how and even why) work with the cloudstack templates.

I think you probably already have a better idea about what does vagrant actually do with that parameter. From everything it does we need to cut some of it off (and delegate to cloudstack) and keep the rest, probably with adaptations.

(Just brainstorming here...) Does that make any sense?